dashboard/lib/contact_rollups_v2.rb

```diff
@@ -29,7 +29,16 @@ def self.build_contact_rollups(log_collector, sync_with_pardot=false)
     end
 
     log_collector.time!("Overwrites contact_rollups_final table") do
-      ContactRollupsFinal.overwrite_from_processed_table
+      truncate_or_delete_table ContactRollupsFinal
+      ContactRollupsFinal.insert_from_processed_table
     end
   end
+
+  # Using truncate allows us to re-use row IDs,
+  # which is important in production so we don't overflow the table.
+  # Deletion is required in test environments, as tests generally do
+  # not allow you to execute TRUNCATE statements.
+  def self.truncate_or_delete_table(model)
+    CDO.rack_env == :production ? model.truncate_table : model.delete_all
```

Comment:
Should we call `ActiveRecord::Base.connection.truncate(model.table_name)` here instead of duplicating implementation of `truncate_table` method in ContactRollupsFinal, ContactRollupsProcessed and ContactRollupsRaw?

Reply:
Yeah, I was ambivalent on this -- I'll change it.